onnxruntime/contrib_ops/cuda/quantization/matmul_nbits.cu

```diff
@@ -288,6 +288,7 @@ bool TryMatMul4Bits(
   if (n % kColsPerThreadBlock != 0 || k % 8 != 0 || m > 1) {
     return false;
   }
+  const int kWarpSize = GPU_WARP_SIZE_HOST;
```

Comment:
Put a #define around this via how we handle things with USE_ROCM and USE_MIGRAPHX

```
#define USE_MIGRAPHX || USE_ROCM
const int kWarpSize = GPU_WARP_SIZE_HOST;
#endif 
```

That way if once we upstream this change it wouldn't break another EP if this is specific to AMD

Reply:
There's no need to ifdef protect this.  CUDA EP also defines GPU_WARP_SIZE_HOST for this reason.

Reply:
Ah then my real curiosity is why redefine it here then if that's the case?